Source/DafnyCore/Compilers/Compiler-java.cs

```diff
@@ -2334,7 +2334,10 @@ private class GenericArrayElementLvalue : ILvalue {
 
     protected string GetClassPath(string targetFilename) {
       var targetDirectory = Path.GetFullPath(Path.GetDirectoryName(targetFilename));
-      return "." + Path.PathSeparator + targetDirectory + Path.PathSeparator + Path.Combine(targetDirectory, "DafnyRuntime.jar");
+      var classpath = Environment.GetEnvironmentVariable("CLASSPATH") ?? "";
+      // Note that the items in the CLASSPATH must have absolute paths because the compilation is performed in a subfolder of where the command-line is executed
+      // Java ignores empty (oand other ill-formed) classpath entries
```

Comment:
Typo: `oand`
But I would just remove this line; it's not ill-formed, it just is the same as `.`

Reply:
Done - though I could not find any documentation that an empty classpath *item* was interpreted as ".", just that anything not a file or directory was ignored.